src/snakemake_software_deployment_plugin_conda/__init__.py

```
@@ -224,6 +248,25 @@ async def deploy(self) -> None:
+                        "pip",
+                        "install",
+                        "--venv",
+                        self.deployment_path,
+                        " ".join(pypi_specs),
+                    ],
+                    check=True,
+                    stdout=sp.PIPE,
+                    stderr=sp.PIPE,
+                )
+            except sp.CalledProcessError as e:
+                raise WorkflowError("Failed to install pypi packages", e)
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)